code/game/turfs/turf.dm

```diff
@@ -637,26 +637,34 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /turf/AllowDrop()
 	return TRUE
 
-/turf/proc/add_vomit_floor(mob/living/M, vomit_type = VOMIT_TOXIC, purge_ratio = 0.1)
+/turf/proc/add_vomit_floor(mob/living/vomiter, vomit_flags, purge_ratio = 0.1)
+	var/obj/effect/decal/cleanable/vomit/throw_up
 
-	var/obj/effect/decal/cleanable/vomit/vomit
-	if (vomit_type == VOMIT_NEBULA)
-		vomit = new /obj/effect/decal/cleanable/vomit/nebula(src, M.get_static_viruses())
+	if (vomit_flags & MOB_VOMIT_COLOR_NEBULA)
+		throw_up = new /obj/effect/decal/cleanable/vomit/nebula(src, vomiter.get_static_viruses())
 	else
-		vomit = new /obj/effect/decal/cleanable/vomit(src, M.get_static_viruses())
+		throw_up = new /obj/effect/decal/cleanable/vomit(src, vomiter.get_static_viruses())
 
-	//if the vomit combined, apply toxicity and reagents to the old vomit
-	if (QDELETED(vomit))
-		vomit = locate() in src
-	if(!vomit)
+	// if the vomit combined, apply toxicity and reagents to the old vomit
+	if (QDELETED(throw_up))
+		throw_up = locate() in src
+	if(isnull(throw_up))
 		return
+
 	// Apply the proper icon set based on vomit type
-	if(vomit_type == VOMIT_PURPLE)
-		vomit.icon_state = "vomitpurp_[pick(1,4)]"
-	else if (vomit_type == VOMIT_TOXIC)
-		vomit.icon_state = "vomittox_[pick(1,4)]"
-	if (purge_ratio && iscarbon(M))
-		clear_reagents_to_vomit_pool(M, vomit, purge_ratio)
+	if(vomit_flags & MOB_VOMIT_COLOR_PURPLE)
+		throw_up.icon_state = "vomitpurp_[pick(1,4)]"
+	else if (vomit_flags * MOB_VOMIT_COLOR_NANITES)
+		throw_up.icon_state = "vomitnanite_[pick(1,4)]"
+	else if (vomit_flags & MOB_VOMIT_COLOR_TOXIC)
+		throw_up.icon_state = "vomittox_[pick(1,4)]"
```

Comment:
The combination of these flags and `MOB_VOMIT_COLOR_NEBULA` are so ugly, is there really no better way we can do this?
Can we not just pass a vomit typepath in here and have done with it?

Reply:
i did something different that should make it a bit cleaner as far as working. i also didn't catch that i did a `*` instead of a `&` there, whoops

Reply:
That's better but I'll be honest I still don't love it.
These are provided as bitflags which would often imply to me that providing `(MOB_VOMIT_COLOR_NEBULA | MOB_VOMIT_COLOR_PURPLE)` would do something different to `(MOB_VOMIT_COLOR_NEBULA)`, which it does not.

Are there any cases in here where the fact that we have a priority order in which we decide which of these to use is important? It didn't look like it to me.
I don't otherwise understand why we would want to populate the bitflag with fields representing different type paths when we could just pass explicitly the type of vomit we want to spawn

Reply:
i don't like the number of args but it's done